### PR TITLE
Upgrade to elixir 1.4 and update outdated hex packages

### PR DESCRIPTION
**Pull request by auxbuss**
This performs an upgrade to elixir 1.4, updates outdated hex packages, and fixes the tests. It also adds `asdf`, otherwise the project fails to build for anyone with `asdf` installed.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -30,19 +31,19 @@ defmodule App.Mixfile do
...
      {:phoenix_pubsub, "~> 1.0"},
      {:phoenix_ecto, "~> 3.0"},
      {:postgrex, ">= 0.0.0"},
      {:phoenix_html, "~> 2.6"},
      {:phoenix_live_reload, "~> 1.0", only: :dev},
      {:gettext, "~> 0.11"},
      {:cowboy, "~> 1.0"},
-     {:absinthe, "~> 1.2.0"},
-     {:absinthe_plug, "~> 1.1"},
-     {:absinthe_ecto, git: "https://github.com/absinthe-graphql/absinthe_ecto.git"},
```
Comment (carleryd) [inline mix.exs:43]: A bit picky maybe but this follows master branch right? I've had issues with this before when the master branch changes and breaks my project even though seemingly nothing has changed. It's more stable to either follow a specific version or commit, is this possible? :)

**Review by carleryd — COMMENTED** [review]
Great stuff! Sorry for the delay. What do you think about the comment I left?

File: mix.exs (hunk 1 above)
Comment (auxbuss) [reply mix.exs:43]: As the `mix.lock` determines package versions, nothing will be updated until a `mix deps.update`, so things should be fine. Still, most things here are fairly fast moving, so breakages are likely, I guess, from time to time.

I'm not clear which package you're referring to, tbh, though.

Regardless, it'd be nice to get some more tests around all of this to lessen the chances of something falling between the cracks.

**Comment by auxbuss** [comment]
I fixed up the conflicts on this branch caused by the npm and hex package updates on master.

File: mix.exs (hunk 1 above)
Comment (carleryd) [reply mix.exs:43]: I'm sorry I missread the diff. And yes it could definitely use some tests!
Thanks for the contribution! 👍

**Pull request merged**